ansible/roles/go-ipfs-deploy-docker/files/dockerfiles/container_daemon

```diff
@@ -0,0 +1,63 @@
+#!/bin/sh
+set -e
+
+envToConfig() {
```

Comment:
we should run the docker container from upstream. This is prone to missing upgrades or security fixes and the replace magic in this file is scary. If env variables are necessary we should bring the issue upstream (but I understand they are not necessary because we have a configuration file anyways).

Reply:
This is already based on an intermediary Docker build that is using the upstream `Dockerfile`

Reply:
But not using upstream entrypoint

Reply:
that's because the upstream entrypoint doesn't support setting config options via ENV.

Reply:
I know, but aren't you mounting a config file too? It's nice to use Env when you got rid of the config file, but now we have:

* env variables
* a config file
* a special entrypoint
* sed/jq magic
* a special docker build of the upstream container
* some variables like [Swarm_AddrFilters](https://github.com/ipfs/infra/pull/487/files/e5af3ead9b46c58afec2942f824958f8de1f7cfa#diff-137d33d0acd1274373c462745e48b669R22) being a single-line string array because I suspect the magic cannot do the right replacement if this was a readable yaml array.
* No way to see differences applied to the configuration when running ansible in `--diff --check` mode.

So I suggest to:

1) Template the config file and drop all custom docker and scripts
2) Follow up on https://github.com/ipfs/go-ipfs/issues/6526 to ensure that env-variable configuration is supported upstream

Reply:
>     * sed/jq magic
 this is needed to work around the config file's strict format (JSON).
awk/jq are not `magic`. they are the tools that are available in the chosen image and their usage is commented, explaining what each invocation does. 

>     * a special docker build of the upstream container
this is a pretty common pattern

>     * some variables like [Swarm_AddrFilters](https://github.com/ipfs/infra/pull/487/files/e5af3ead9b46c58afec2942f824958f8de1f7cfa#diff-137d33d0acd1274373c462745e48b669R22) being a single-line string array because I suspect the magic cannot do the right replacement if this was a readable yaml array.

check the latest commit, I've added support for multi-line ENVs

>     2. Follow up on [ipfs/go-ipfs#6526](https://github.com/ipfs/go-ipfs/issues/6526) to ensure that env-variable configuration is supported upstream
This was already rejected once https://github.com/ipfs/go-ipfs/pull/6325 

While this is not the perfect solution, it unblocks us in our pursuit of streamlined deployments and it allows us to start collecting metrics from go-ipfs. 

Please feel free to contribute a PR with your suggested changes.

Reply:
> they are the tools that are available in the chosen image

you need to install `jq` explicitally because it's not part of the image

> this is a pretty common pattern

A common pattern when you work with a black box. We are part of the ipfs project. We are the same team. There are alternatives to this approach to allow env-var overwrites of configuration options.

> This was already rejected once ipfs/go-ipfs#6325

It was rejected because it's not the right way of doing it (plenty of reasons in there). Putting it here instead does not make it any better.

> check the latest commit, I've added support for multi-line ENVs

It's a single string, reads better but still not an actual array.

> This unblocks us in our pursuit of streamlined deployments and it allows us to start collecting metrics from go-ipfs.

This is unrelated to collecting metrics and does not block docker deployments (if anything simplifies them).

> Please feel free to contribute a PR with your suggested changes.

The fact that a custom entrypoint with jq / sed ad-hoc replacements causes someone to raise flags twice on unrelated occasions should make you think whether this is the right approach.